src/providers/irc/IrcConnection2.cpp

```diff
@@ -13,40 +17,96 @@ namespace {
 IrcConnection::IrcConnection(QObject *parent)
     : Communi::IrcConnection(parent)
 {
-    // send ping every x seconds
+    // Log connection errors for ease-of-debugging
+    QObject::connect(this, &Communi::IrcConnection::socketError, this,
+                     [this](QAbstractSocket::SocketError error) {
+                         qCDebug(chatterinoIrc) << "Connection error:" << error;
+                     });
+
+    QObject::connect(
+        this, &Communi::IrcConnection::socketStateChanged, this,
+        [this](QAbstractSocket::SocketState state) {
+            if (state == QAbstractSocket::UnconnectedState)
+            {
+                if (!this->expectConnectionLoss_.load())
+                {
+                    // Ensure we always notify our parent that we should
```

Comment:
Could you expand on the comments here explaining under what circumstances we wouldn't want to call connectionLost here?